cli/src/commands/squash.rs

```diff
@@ -125,98 +131,108 @@ pub fn move_diff(
-        "\
-You are moving changes from: {}
-into commit: {}
-
-The left side of the diff shows the contents of the parent commit. The
-right side initially shows the contents of the commit you're moving
-changes from.
-
-Adjust the right side until the diff shows the changes you want to move
-to the destination. If you don't make any changes, then all the changes
-from the source will be moved into the destination.
-",
-        tx.format_commit_summary(&source),
-        tx.format_commit_summary(&destination)
-    );
-    let new_parent_tree_id =
-        diff_selector.select(&parent_tree, &source_tree, matcher, Some(&instructions))?;
-    if new_parent_tree_id == parent_tree.id() {
-        if diff_selector.is_interactive() {
-            return Err(user_error("No changes selected"));
-        }
+        .check_rewritable(sources.iter().chain([&destination]))?;
+    // Tree diffs to apply to the destination
+    let mut tree_diffs = vec![];
+    // Commits to include descriptions from
+    let mut description_sources = vec![&destination];
+    for source in &sources {
+        let parent_tree = merge_commit_trees(tx.repo(), &source.parents())?;
+        let source_tree = source.tree()?;
+        let instructions = format!(
+            "\
+    You are moving changes from: {}
+    into commit: {}
+    
+    The left side of the diff shows the contents of the parent commit. The
+    right side initially shows the contents of the commit you're moving
+    changes from.
+    
+    Adjust the right side until the diff shows the changes you want to move
+    to the destination. If you don't make any changes, then all the changes
+    from the source will be moved into the destination.
+    ",
+            tx.format_commit_summary(source),
+            tx.format_commit_summary(&destination)
+        );
+        let new_parent_tree_id =
+            diff_selector.select(&parent_tree, &source_tree, matcher, Some(&instructions))?;
+        if new_parent_tree_id == parent_tree.id() {
+            if diff_selector.is_interactive() {
+                return Err(user_error("No changes selected"));
+            }
 
-        if let [only_path] = path_arg {
-            if no_rev_arg
-                && revset::parse(
-                    only_path,
-                    &tx.base_workspace_helper().revset_parse_context(),
-                )
-                .is_ok()
-            {
-                writeln!(
-                    ui.warning(),
-                    "warning: The argument {only_path:?} is being interpreted as a path. To \
-                     specify a revset, pass -r {only_path:?} instead."
-                )?;
+            if let [only_path] = path_arg {
+                if no_rev_arg
+                    && revset::parse(
+                        only_path,
+                        &tx.base_workspace_helper().revset_parse_context(),
+                    )
+                    .is_ok()
+                {
+                    writeln!(
```

Comment:
Perhaps, this error should be printed only once (when no changes are selected at all?)

I'm not sure about the interactive case, but it might be okay to skip some commits by not selecting changes.